src/System.Buffers.ReaderWriter/System/Buffers/Reader/BufferReader_search.cs

```diff
@@ -427,7 +429,7 @@ public unsafe bool TryReadTo(out ReadOnlySequence<T> sequence, ReadOnlySpan<T> d
             }
             else
             {
-                peekBuffer = new Span<T>(new T[delimiter.Length]);
+                peekBuffer = new Span<T>(_arrayPool.Rent(delimiter.Length));
```

Comment:
How are you going to return this array if the only reference to it is the one stored inside the span?

Reply:
Not going to return it as there is nothing to return to.